File: bus-booking-frontend/src/services/busService.js
import api from "../api/axiosInstance";

export const getAllBuses  = ()       =>
  api.get("/buses");

export const getBusById  = (id)     =>
  api.get(`/buses/${id}`);

export const addBus      = (data)   =>
  api.post("/buses", data);

export const updateBus   = (id, d)  =>
  api.put(`/buses/${id}`, d);

export const deleteBus   = (id)     =>
  api.delete(`/buses/${id}`);

File: BusTicketBookingSystem/src/main/java/com/ash/bbus/web/BusTicketBookingSystem/service/BusService.java
package com.ash.bbus.web.BusTicketBookingSystem.service;

import java.util.List;

import com.ash.bbus.web.BusTicketBookingSystem.dto.BusDTO;
import com.ash.bbus.web.BusTicketBookingSystem.dto.SeatDTO;

public interface BusService {

	BusDTO addBus(BusDTO dto);

	BusDTO getBusById(Long busId);

	List<BusDTO> getAllBuses();

	BusDTO updateBus(Long busId, BusDTO dto);

	void deleteBus(Long busId);

	void addSeatsToBus(Long busId, List<SeatDTO> seats);

	List<SeatDTO> getSeatsByBus(Long busId);
}